### PR TITLE
[FLINK-9674][tests] Replace hard-coded sleeps in QS E2E test 

### PR DESCRIPTION
**Pull request by zentol**
## What is the purpose of the change

This PR replaces a hard-coded sleep in a QueryableState end-to-end test. The test was sleeping for 65 seconds after the TM was restarted, to wait until the restarted.

Instead we now first wait for the job restart procedure to kick in (using a new method to check for state transitions) and then waiting for the job to run.

Additionally
* reduce `heartbeat.timeout` to speed up TM loss detection
* fix `common#wait_job_running` returning true for scheduled jobs, we now pass the `-r` flag
* simplify jar paths in QS test (now relative to END_TO_END_DIR)
* change `test-runner-common#cleanup` to use existing `common#clean_[log|stdout]_files` functions

## Verifying this change

* manually verified

### CODE REVIEW

**Comment by yanghua** [comment]
+1

File: flink-end-to-end-tests/test-scripts/test_queryable_state_restart_tm.sh [hunk 1]
```diff
@@ -85,20 +90,23 @@ function run_test() {
...
 
-    sleep 65 # this is a little longer than the heartbeat timeout so that the TM is gone
+    # wait until the TM loss was detected
+    wait_for_job_state_transition ${JOB_ID} "RESTARTING" "CREATED"
 
     start_and_wait_for_tm
 
+    wait_job_running ${JOB_ID}
+
+    local current_num_checkpoints="$(get_completed_number_of_checkpoints ${JOB_ID})"
```
Comment (florianschmidt1994) [inline flink-end-to-end-tests/test-scripts/test_queryable_state_restart_tm.sh]: Why did you move this from `before killing the TM` to `after having the TM restarted again` ?

**Comment by zentol** [comment]
@florianschmidt1994 I don't see the comment you're referring to :(
(Did you start a review, but wrote your comment separately? Then your review would still be in progress!)

The issues you identified are certainly valid, but I'm wondering if it really makes sense to invest time into this now.
There've been discussions about writing a python/java based framework for the tests (and I've already started tinkering on a java version); so any _significant_ changes we make to the bash-scripts might be subsumed soon&#8482;.

**Comment by florianschmidt1994** [comment]
> (Did you start a review, but wrote your comment separately? Then your review would still be in progress!)

Yes :D I submitted it now

>The issues you identified are certainly valid, but I'm wondering if it really makes sense to invest time into this now.
>There've been discussions about writing a python/java based framework for the tests (and I've already started tinkering on a java version); so any significant changes we make to the bash-scripts might be subsumed soon™.

Alright, then let's leave it this way and we can still take care of it should we run into problems with this in the future

File: flink-end-to-end-tests/test-scripts/test_queryable_state_restart_tm.sh (hunk 1 above)
Comment (zentol) [reply flink-end-to-end-tests/test-scripts/test_queryable_state_restart_tm.sh]: So when I ran this test locally it frequently occurred that we never actually waited for checkpoints to complete; the number of checkpoint that we waited for had already occurred and we exited early.
The job just switches to running, and right away we got 3 completed checkpoints?
This was a bit, well, _odd_. 

In the previous version the checkpoint count could further increase while we shut down the TM.
Technically there's no guarantee how up-to-date the result if `get_completed_number_of_checkpoints` or how long `kill_random_taskmanager` actually takes, so we may have fulfilled the checkpoint count condition before the job even restarts. It's admittedly unlikely.

This change simply guarantees that the job actually completes N checkpoints after it was restarted, and just serves to eliminate doubts about the correctness of the test.

**Comment by zentol** [comment]
merging.
